rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -93,6 +94,41 @@
 MIN_LOGS_PROCESSED_TO_SLEEP = 1000
 
 
+def decoding_wrapper(
+        msg_aggregator: 'MessagesAggregator | None',
+        transaction_for_logs: EvmTransaction,
+        func: Callable,
+        *args: tuple[Any],
+        **kwargs: Any,
+) -> tuple[Any, bool]:
+    """
+    Wrapper for methods that execute logic from decoders. It handles handles the errors logging
+    them and optionally sending them to the user.
+
+    It returns a tuple where the first argument is the output of func and the second is a boolean
+    set to True if an error was raised from func.
+    """
+    try:
+        return func(*args, **kwargs), False
+    except (
+        UnknownAsset,
+        WrongAssetType,
+        DeserializationError,
+        IndexError,
+        ValueError,
+        ConversionError,
+    ) as e:
+        log.error(traceback.format_exc())
+        log.error(
+            f'Decoding of transactions {transaction_for_logs} failed due to {e} when calling '
+            f'{func.__name__} with {args} {kwargs}',
+        )
+        if msg_aggregator is not None:
```

Comment:
So the code you wrote does not use this at all. You always set it to `None`. So this also changes current behaviour where it's done 2/4 places you touched.

But here we need to be consistent. We either always send a message to the user or never. So I would get rid of the if or of it alltogether.

Need to think when we need this to tell the users something went wrong. I think the answer is always? The code before your change did it in 2/4 places from what I can see. But perhaps best is to always do it? So get rid of the if. What do you think?